pkg/infrastructure/azure/storage.go

```diff
@@ -292,21 +283,15 @@ func CreatePageBlob(ctx context.Context, in *CreatePageBlobInput) (string, error
 			Metadata: metadata,
 		})
 		if err != nil {
-			return "", fmt.Errorf("failed to create page blob with image URL: %w", err)
+			return fmt.Errorf("failed to create page blob with image URL: %w", err)
 		}
 
 		err = doUploadPagesFromURL(ctx, pageBlobClient, in.ImageURL, in.ImageLength)
 		if err != nil {
-			return "", fmt.Errorf("failed to upload page blob image from URL %s: %w", in.ImageURL, err)
+			return fmt.Errorf("failed to upload page blob image from URL %s: %w", in.ImageURL, err)
 		}
 	}
-
-	// Is this addition OK for when CreatePageBlob() is called from InfraReady()
-	sasURL, err := pageBlobClient.GetSASURL(sas.BlobPermissions{Read: true}, time.Now().Add(time.Minute*60), &blob.GetSASURLOptions{})
-	if err != nil {
-		return "", fmt.Errorf("failed to get Page Blob SAS URL: %w", err)
-	}
```

Comment:
I was surprised to see this function (I didn't know or forgot about it), but you are correct to remove it. The function docs say it depends on shared-key credentials.

Reply:
Actually we may want to keep this, and make it conditional for azure stack